Sources/TuistGenerator/Generator/SchemesGenerator.swift

```diff
@@ -119,9 +119,24 @@ final class SchemesGenerator: SchemesGenerating {
         if fileHandler.exists(sharedPath) { try fileHandler.delete(sharedPath) }
     }
     
-    private func createDefaultScheme(target: Target, project: Project, buildConfiguration: String) -> Scheme {
+    private func createDefaultScheme(target: Target, project: Project, buildConfiguration: String, graph: Graphing) -> Scheme {
         let targetReference = TargetReference.project(path: project.path, target: target.name)
-        let testTargets = target.product.testsBundle ? [TestableTarget(target: targetReference)] : []
+        
+        let testTargets: [TestableTarget]
+        
+        if target.product.testsBundle {
+            testTargets = [TestableTarget(target: targetReference)]
+        } else if let targetNode = graph.findTargetNode(path: project.path, name: target.name) {
+            // Find all test targets that depend on this target
+            testTargets = graph.targets
```

Comment:
As a suggestion, we can limit the scope of the search to targets within the same project as opposed to the whole workspace

A few ways we can go about it:

1) Scan project targets directly

```swift
targetNode.project
          .targets
          .filter { $0.product.testsBundle }
          .filter { $0.dependencies.contains(.target(name: targetNode.name)) }
          .map { TargetReference.project(path: targetNode.project.path, target: $0.name) }
// ...
```

2) Add another helper to the graph

In graph we could introduce another method:

```swift
/// Returns all target nodes at a given path (i.e. all target nodes in a project)
public func targetNodes(at path: AbsolutePath) -> [TargetNode] {
        guard let nodes = cache.targetNodes[path] else {
            return []
        }
        return Array(nodes.values)
    }
```

Reply:
Makes sense, I was thinking what could be the performance impact of looping through all targets. Only looking through project targets sounds much better.

Reply:
@kwridan I added the helper, might be useful in other places too